server/google_resources.py

```diff
@@ -23,6 +24,34 @@
 API_SERVICE_NAME = 'gmail'
 API_VERSION = 'v1'
 
+def login_required(f):
```

Comment:
you shouldn't need to copy this in, you can just use `from resources import login_required` for now and we can move functions like this to a utils.py or something later.